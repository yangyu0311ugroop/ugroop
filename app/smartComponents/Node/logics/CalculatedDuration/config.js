import { NODE_API, UPDATE_NODE } from 'apis/constants';
import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';

export const CONFIG = {
  value: {
    duration: ({ templateId }) =>
      NODE_STORE_SELECTORS.duration({ id: templateId }),
    childrenCount: ({ tabId }) =>
      NODE_STORE_SELECTORS.childrenCount({ id: tabId }),
  },
  setValue: {},

  isLoading: {
    updatingNode: [NODE_API, UPDATE_NODE],
  },
};
