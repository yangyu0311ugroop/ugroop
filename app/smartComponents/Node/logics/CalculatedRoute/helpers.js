import dotProp from 'dot-prop-immutable';

let fetchedRoutes = {
  test: 'test',
  testMerge: { id: 2233 },
};

const getCache = (prop, value) =>
  prop && dotProp.get(fetchedRoutes, prop, value);
const setCache = (prop, value) => {
  fetchedRoutes = dotProp.set(fetchedRoutes, `${prop}`, value);
};
const mergeCache = (prop, value) => {
  fetchedRoutes = dotProp.merge(fetchedRoutes, prop, value);
};
const deleteCache = routeId => {
  fetchedRoutes = dotProp.delete(fetchedRoutes, `${routeId}`);
};
const isValid = raw => {
  if (!Array.isArray(raw) || !raw.length) {
    return false;
  }

  for (let i = 0; i < raw.length; i += 1) {
    if (!raw[i]) return false;
  }

  return true;
};

export const ROUTE_CACHE = {
  fetchedRoutes,
  delete: deleteCache,
  get: getCache,
  set: setCache,
  merge: mergeCache,
  isValid,
};
