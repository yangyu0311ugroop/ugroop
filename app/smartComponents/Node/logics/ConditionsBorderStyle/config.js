import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';

export const CONFIG = {
  value: {
    calculatedMedicalSeverity: NODE_STORE_SELECTORS.calculatedMedicalSeverity,
    calculatedMedicalCount: NODE_STORE_SELECTORS.calculatedMedicalCount,
    calculatedDietaryCount: NODE_STORE_SELECTORS.calculatedDietaryCount,
  },
};
