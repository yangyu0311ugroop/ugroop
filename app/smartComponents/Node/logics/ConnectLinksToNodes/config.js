import {
  LINK_STORE_SELECTORS,
  NODE_STORE_SELECTORS,
} from 'datastore/nodeStore/selectors';
import { USER_STORE_SELECTORS } from 'datastore/userStore/selectors';

export const GET_USER_NODE = {
  value: {
    userNode: LINK_STORE_SELECTORS.userNode,
  },
};

export const GET_USER_NODE_CONTENT = {
  value: {
    content: ({ userNode }) =>
      NODE_STORE_SELECTORS.content({ id: userNode[0] }),
  },
};

export const CONFIG = {
  value: {
    knownAs: ({ content }) => USER_STORE_SELECTORS.knownAs({ id: content }),
  },
  setValue: {},
};
