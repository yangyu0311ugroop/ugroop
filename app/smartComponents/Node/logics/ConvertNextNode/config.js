import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';

export const CONFIG = {
  value: {
    nextNodeId: NODE_STORE_SELECTORS.firstNextNodeId,
  },
  setValue: {
    checklists: NODE_STORE_SELECTORS.parentChecklists,
    nextNodes: NODE_STORE_SELECTORS.nextNodes,
  },
};
