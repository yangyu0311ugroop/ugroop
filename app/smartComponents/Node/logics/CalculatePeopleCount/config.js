import {
  TOUR_CONTRIBUTOR_ROLE,
  TOUR_CONTRIBUTOR_ROLE_TYPES,
} from 'apis/components/Ability/roles';
import { INVITATION_STORE_SELECTORS } from 'datastore/invitationStore/selectors';
import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { COGNITO_STORE_SELECTORS } from 'datastore/stormPathStore/selectors';
import { TEMPLATE_MANAGEMENT_STORE_SELECTORS } from 'datastore/templateManagementStore/selectors';
import { USER_STORE_SELECTORS } from 'datastore/userStore/selectors';
import { RESAGA_HELPERS } from 'utils/helpers/resaga';
import { UID_HELPERS } from 'utils/helpers/uid';
import dropRight from 'lodash/dropRight';
import union from 'lodash/union';

export const CONFIG_0 = {
  value: {
    templateId: TEMPLATE_MANAGEMENT_STORE_SELECTORS.templateId,
    userId: COGNITO_STORE_SELECTORS.myId,
    calculatedPeopleCount: NODE_STORE_SELECTORS.calculatedPeopleCount,
  },
};

export const GET_ROLES = {
  value: {
    roles: {
      getter: ({ roles = [] }) => {
        if (
          roles.includes(TOUR_CONTRIBUTOR_ROLE.TOUR_ORGANIZER) ||
          roles.includes(TOUR_CONTRIBUTOR_ROLE.TOUR_OWNER) ||
          roles.includes(TOUR_CONTRIBUTOR_ROLE.TOUR_COLLABORATOR) ||
          roles.includes(TOUR_CONTRIBUTOR_ROLE.TOUR_VIEWER)
        ) {
          return TOUR_CONTRIBUTOR_ROLE_TYPES;
        }
        // for non contributors and public
        return [
          TOUR_CONTRIBUTOR_ROLE.TOUR_ORGANIZER,
          TOUR_CONTRIBUTOR_ROLE.TOUR_OWNER,
        ];
      },
    },
  },
};

export const CONFIG = {
  value: {
    ownerId: RESAGA_HELPERS.mapToId(
      NODE_STORE_SELECTORS.createdBy,
      'templateId',
    ),

    followersEmail: {
      keyPath: ({ followers = [] }) =>
        followers.map(followerId =>
          NODE_STORE_SELECTORS.email({ id: followerId }),
        ),
      cacheKey: ({ followers = [] }) =>
        `CalculatePeopleCount.${followers.toString()}.followersCount`,
      props: ({ followers }) => followers,
      getter: (...args) => {
        const followerEmails = dropRight(args);

        return followerEmails.filter(email => email !== undefined);
      },
    },

    participantEmails: {
      keyPath: ({ participants = [] }) =>
        participants.map(participantId =>
          NODE_STORE_SELECTORS.email({ id: participantId }),
        ),
      cacheKey: ({ participants = [] }) =>
        `CalculatePeopleCount.${participants.toString()}.participantsCount`,
      props: ({ participants }) => participants,
      getter: (...args) => {
        const participantEmails = dropRight(args);

        return participantEmails.filter(email => email !== undefined);
      },
    },

    organiserUserIds: {
      keyPath: ({ userNodeIds = [] }) =>
        userNodeIds.map(userNodeId =>
          INVITATION_STORE_SELECTORS.userNodeUserId({ id: userNodeId }),
        ),
      cacheKey: ({ userNodeIds = [] }) =>
        `CalculatePeopleCount.${userNodeIds.toString()}.organiserUserIds`,
      props: ({ userNodeIds }) => userNodeIds,
      getter: (...args) => {
        const organiserEmails = dropRight(args);

        return organiserEmails;
      },
    },
  },
};

export const CONFIG_2 = {
  value: {
    ownerEmail: RESAGA_HELPERS.mapToId(USER_STORE_SELECTORS.email, 'ownerId'),
    organiserEmails: {
      keyPath: ({ organiserUserIds = [] }) =>
        organiserUserIds.map(organiserUserId =>
          USER_STORE_SELECTORS.email({ id: organiserUserId }),
        ),
      cacheKey: ({ organiserUserIds = [] }) =>
        `CalculatePeopleCount.${organiserUserIds.toString()}.organiserEmails`,
      props: ({ organiserUserIds }) => organiserUserIds,
      getter: (...args) => {
        const organiserEmails = dropRight(args);

        return organiserEmails.filter(email => email !== undefined);
      },
    },
  },
};

export const CONFIG_3 = {
  value: {
    mergedUserEmails: {
      cacheKey: ({ followers = [], userNodeIds = [], participants = [] }) =>
        `CalculatePeopleCount.${followers.toString()}.${userNodeIds.toString()}.${participants.toString()}.mergedUserEmails`,
      getter: ({
        organiserEmails = [],
        participantEmails = [],
        followersEmail = [],
        ownerEmail,
      }) => {
        const filteredFollowerEmails = followersEmail.map(email =>
          email === '' ? UID_HELPERS.generateUID() : email,
        );

        const filteredParticipantEmails = participantEmails.map(email =>
          email === '' ? UID_HELPERS.generateUID() : email,
        );

        const filteredOrganiserEmails = organiserEmails.map(email =>
          email === '' ? UID_HELPERS.generateUID() : email,
        );

        const merged = union(
          filteredFollowerEmails,
          filteredParticipantEmails,
          filteredOrganiserEmails,
          [ownerEmail],
        );

        return merged;
      },
    },
  },
  setValue: {
    calculatedPeopleCount: NODE_STORE_SELECTORS.calculatedPeopleCount,
  },
};
