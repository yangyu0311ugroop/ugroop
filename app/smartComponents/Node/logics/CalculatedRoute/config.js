import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { ROUTE_HELPERS } from 'smartComponents/Node/types/Route/helpers';
import { RESAGA_HELPERS } from 'utils/helpers/resaga';

const trim = params => {
  const { placeIds: places, routeIds: routes } = params;

  // remove nodes that doesn't have placeIds
  return {
    ...params,
    ...routes.reduce(
      ({ placeIds, routeIds, trimmedIds }, id, index) => {
        if (!places[index])
          return {
            placeIds,
            routeIds,
            trimmedIds: trimmedIds.concat(id),
          };

        return {
          placeIds: placeIds.concat(places[index]),
          routeIds: routeIds.concat(id),
          trimmedIds,
        };
      },
      { placeIds: [], routeIds: [], trimmedIds: [] },
    ),
  };
};

const selectRouteIds = {
  cacheKey: ({ routeIds = [] }) => `routes.${routeIds.toString()}.placeIds`,
  keyPath: ({ routeIds = [] }) =>
    routeIds.map(id => NODE_STORE_SELECTORS.placeId({ id })),
  props: RESAGA_HELPERS.prop('routeIds'),
  getter: (...placeIds) => {
    const routeIds = placeIds.pop();

    if (!Array.isArray(routeIds) || !routeIds.length) return {};

    return CONFIG_HELPERS.trim({ placeIds, routeIds });
  },
  spreadObject: true,
};

export const CONFIG_HELPERS = {
  trim,
  selectRouteIds,
};

export const CONFIG = {
  value: {
    routeProps: {
      cacheKey: ({ origin, destination }) => `${origin}_${destination}.route`,
      keyPath: [
        ({ parentId }) => NODE_STORE_SELECTORS.children({ id: parentId }),
        NODE_STORE_SELECTORS.origin,
        NODE_STORE_SELECTORS.destination,
      ],
      props: [
        RESAGA_HELPERS.prop('origin'),
        RESAGA_HELPERS.prop('destination'),
      ],
      getter: (
        ids,
        originState,
        destinationState,
        originProp,
        destinationProp,
      ) => {
        let origin;
        let destination;

        if (originProp && destinationProp) {
          origin = originProp;
          destination = destinationProp;
        } else {
          origin = originState;
          destination = destinationState;
        }

        return {
          origin,
          destination,
          routeIds: ROUTE_HELPERS.routeIds({ origin, destination, ids }),
        };
      },
      spreadObject: true,
    },
  },
};

export const PLACE_IDS_CONFIG = {
  value: {
    placeIds: CONFIG_HELPERS.selectRouteIds,
    currentPlaceIds: NODE_STORE_SELECTORS.calculatedPlaceIds,
    // currentRoutes: NODE_STORE_SELECTORS.calculatedRoutes,
  },
  setValue: {
    placeIds: NODE_STORE_SELECTORS.calculatedPlaceIds,
    routeIds: NODE_STORE_SELECTORS.calculatedRouteIds,
    // routes: NODE_STORE_SELECTORS.calculatedRoutes,
    routeError: NODE_STORE_SELECTORS.calculatedRouteError,
  },
};
