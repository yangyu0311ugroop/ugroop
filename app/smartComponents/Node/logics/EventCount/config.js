import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { NODE_STORE_UTILS } from 'datastore/nodeStore/utils';
import { EVENT_GROUPINGS } from 'utils/constants/events';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import { DEFAULT, EVENT_STORE } from 'appConstants';
import { compact } from 'lodash';
export const eventView = [
  'ITINERARY',
  'ACCOMMODATIONS',
  'ACTIVITIES',
  'FOOD',
  'TRANSPORTATION',
];

export const grouping = headerValue =>
  LOGIC_HELPERS.switchCase(headerValue, {
    ACCOMMODATIONS: EVENT_GROUPINGS.accommodationGrouping,
    TRANSPORTATION: EVENT_GROUPINGS.transportationGrouping,
    FOOD: EVENT_GROUPINGS.foodGrouping,
    ATTACHMENT: EVENT_GROUPINGS.allEvents,
    [DEFAULT]: EVENT_GROUPINGS.activityGrouping,
  });

export const CONFIG = {
  value: {
    eventValues: {
      keyPath: ({ ids }) =>
        ids.map(id => NODE_STORE_SELECTORS.calculatedEvents({ id })),
      cacheKey: ({ ids = [] }) =>
        `eventCount.eventValues.${ids.toString()}.overview.filter`,
      getter: (...values) => {
        values.pop();
        return values.map(value => value);
      },
    },
    sectionValues: {
      keyPath: ({ ids }) =>
        ids.map(id => NODE_STORE_SELECTORS.children({ id })),
      cacheKey: ({ ids = [] }) =>
        `eventCount.sectionValues.${ids.toString()}.overview.filter`,
      getter: (...values) => {
        values.pop();
        return values.map(value => value);
      },
    },
    eventNodeValues: {
      keyPath: ({ ids }) =>
        ids.map(id => NODE_STORE_SELECTORS.eventIds({ id })),
      cacheKey: ({ ids = [] }) =>
        `eventCount.eventNodeValues.${ids.toString()}.overview.filter`,
      getter: (...values) => {
        values.pop();
        return values.map(value => value);
      },
    },
    checklistValues: {
      keyPath: ({ ids }) =>
        ids.map(id => NODE_STORE_SELECTORS.checklists({ id })),
      cacheKey: ({ ids = [] }) =>
        `eventCount.checklistValues.${ids.toString()}.overview.filter`,
      getter: (...values) => {
        values.pop();
        return values.map(value => value);
      },
    },
    events: {
      getter: ({ eventValues = [] }) =>
        compact(eventValues).reduce((accu, curr) => [...accu, ...curr], []),
    },
    sectionIds: {
      getter: ({ sectionValues = [] }) =>
        compact(sectionValues).reduce((accu, curr) => [...accu, ...curr], []),
    },
    eventNodeIds: {
      getter: ({ eventNodeValues = [] }) =>
        compact(eventNodeValues).reduce((accu, curr) => [...accu, ...curr], []),
    },
    checklistIds: {
      getter: ({ checklistValues = [] }) =>
        compact(checklistValues).reduce((accu, curr) => [...accu, ...curr], []),
    },
  },
};
export const CONFIG1 = {
  value: {
    eventDataIds: {
      keyPath: ({ eventNodeIds }) =>
        eventNodeIds.map(id => NODE_STORE_SELECTORS.eventDataId({ id })),
      cacheKey: ({ eventNodeIds = [] }) =>
        `eventCount.eventDataIds.${eventNodeIds.toString()}.overview.filter`,
      getter: (...values) => {
        values.pop();
        return values.filter(value => !!value);
      },
    },
  },
};

export const CONFIG2 = {
  value: {
    sectionAttachment: {
      keyPath: ({ sectionIds }) =>
        sectionIds.map(id => NODE_STORE_SELECTORS.attachmentId({ id })),
      cacheKey: ({ sectionIds = [] }) =>
        `eventCount.sectionAttachments.${sectionIds.toString()}.overview.filter`,
      getter: (...values) => {
        values.pop();
        return values.filter(value => !!value);
      },
    },
    eventAttachment: {
      keyPath: ({ eventDataIds }) =>
        eventDataIds.map(id => [EVENT_STORE, 'events', id, 'eventAttachments']),
      getter: (...values) => {
        values.pop();
        return compact(values).reduce((accu, curr) => [...accu, ...curr], []);
      },
    },
  },
};

export const CONFIG3 = {
  value: {
    eventCounts: {
      getter: ({
        events = [],
        eventAttachment = [],
        sectionAttachment = [],
        checklistIds = [],
      }) => {
        const eventsData = eventView.reduce(
          (acc, curr) => ({
            ...acc,
            [curr]: events.filter(
              NODE_STORE_UTILS.calculatedEventsFilter({
                grouping: grouping(curr),
                hasTime: 'all',
              }),
            ).length,
          }),
          {},
        );
        return {
          ...eventsData,
          ATTACHMENT: compact(eventAttachment.concat(sectionAttachment)).length,
          checklist: checklistIds.length,
        };
      },
    },
  },
};
