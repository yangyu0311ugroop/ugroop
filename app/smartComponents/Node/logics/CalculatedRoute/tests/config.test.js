import { NODE_STORE_SELECTORS } from 'datastore/nodeStore/selectors';
import { ROUTE_HELPERS } from 'smartComponents/Node/types/Route/helpers';
import { TEST_HELPERS } from 'utils/helpers/testHelpers';
import { CONFIG, CONFIG_HELPERS } from '../config';

describe('CalculatedPlaceIds/config.js', () => {
  afterEach(() => jest.clearAllMocks());

  describe('Smoke Test', () => {
    it('should exists', () => {
      expect(typeof CONFIG).toBe('object');
      expect(typeof CONFIG_HELPERS).toBe('object');
    });
  });

  describe('trim', () => {
    it('should exists', () => {
      expect(
        CONFIG_HELPERS.trim({
          placeIds: ['111', undefined, '333'],
          routeIds: [1, 2, 3],
        }),
      ).toMatchSnapshot();
    });
  });

  describe('routeProps', () => {
    it('should cacheKey', () => {
      expect(
        CONFIG.value.routeProps.cacheKey({
          origin: 112,
          destination: 223,
        }),
      ).toMatchSnapshot();
    });

    it('should keyPath', () => {
      expect(
        CONFIG.value.routeProps.keyPath[0]({
          parentId: 112,
        }),
      ).toEqual(NODE_STORE_SELECTORS.children({ id: 112 }));
    });

    it('should getter prop', () => {
      ROUTE_HELPERS.routeIds = jest.fn(() => [1, 2, 3]);

      expect(
        CONFIG.value.routeProps.getter([1, 2, 3], 2, 3, 4, 5),
      ).toMatchSnapshot();
    });

    it('should getter state', () => {
      ROUTE_HELPERS.routeIds = jest.fn(() => [1, 2, 3]);

      expect(CONFIG.value.routeProps.getter([1, 2, 3], 2, 3)).toMatchSnapshot();
    });
  });

  describe('CONFIG_HELPERS.selectRouteIds', () => {
    it('should cacheKey', () => {
      expect(CONFIG_HELPERS.selectRouteIds.cacheKey({})).toMatchSnapshot();
      expect(
        CONFIG_HELPERS.selectRouteIds.cacheKey({
          routeIds: [1, 2],
        }),
      ).toMatchSnapshot();
    });

    it('should keyPath', () => {
      expect(CONFIG_HELPERS.selectRouteIds.keyPath({})).toMatchSnapshot();
      expect(
        CONFIG_HELPERS.selectRouteIds.keyPath({
          routeIds: [1, 2],
        }),
      ).toMatchSnapshot();
    });

    it('should return {}', () => {
      expect(CONFIG_HELPERS.selectRouteIds.getter()).toEqual({});
      expect(CONFIG_HELPERS.selectRouteIds.getter([])).toEqual({});
    });

    it('should return trim', () => {
      CONFIG_HELPERS.trim = jest.fn(() => 'trim');

      expect(CONFIG_HELPERS.selectRouteIds.getter([1])).toEqual('trim');

      TEST_HELPERS.expectCalled(CONFIG_HELPERS.trim);
    });
  });
});
