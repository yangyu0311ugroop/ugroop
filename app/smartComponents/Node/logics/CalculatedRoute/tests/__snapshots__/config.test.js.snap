// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatedPlaceIds/config.js CONFIG_HELPERS.selectRouteIds should cacheKey 1`] = `"routes..placeIds"`;

exports[`CalculatedPlaceIds/config.js CONFIG_HELPERS.selectRouteIds should cacheKey 2`] = `"routes.1,2.placeIds"`;

exports[`CalculatedPlaceIds/config.js CONFIG_HELPERS.selectRouteIds should keyPath 1`] = `Array []`;

exports[`CalculatedPlaceIds/config.js CONFIG_HELPERS.selectRouteIds should keyPath 2`] = `
Array [
  Array [
    "nodeStore",
    "nodes",
    1,
    "customData",
    "placeId",
  ],
  Array [
    "nodeStore",
    "nodes",
    2,
    "customData",
    "placeId",
  ],
]
`;

exports[`CalculatedPlaceIds/config.js routeProps should cacheKey 1`] = `"112_223.route"`;

exports[`CalculatedPlaceIds/config.js routeProps should getter prop 1`] = `
Object {
  "destination": 5,
  "origin": 4,
  "routeIds": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`CalculatedPlaceIds/config.js routeProps should getter state 1`] = `
Object {
  "destination": 3,
  "origin": 2,
  "routeIds": Array [
    1,
    2,
    3,
  ],
}
`;

exports[`CalculatedPlaceIds/config.js trim should exists 1`] = `
Object {
  "placeIds": Array [
    "111",
    "333",
  ],
  "routeIds": Array [
    1,
    3,
  ],
  "trimmedIds": Array [
    2,
  ],
}
`;
